xbmc/cores/FFmpeg.h

```diff
@@ -51,3 +74,7 @@ class CFFmpegLog
   int level;
 };
 
+int GetPacketExtradata(const AVPacket* pkt,
```

Comment:
You could return `uint8_t*` (which is the allocated extradata), That way you can avoid the double pointer. You can use the address for the size `int& extraDataSize` or just return a `std::tuple<uint8_t*, int>`.